example/src/main/java/com/ryanharter/auto/value/gson/example/BirthdateAdapter.java

```diff
@@ -0,0 +1,44 @@
+package com.ryanharter.auto.value.gson.example;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+public class BirthdateAdapter extends TypeAdapter<Date> {
+  private final DateFormat df = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
+
+  @Override public void write(JsonWriter out, Date value) throws IOException {
+    if (value == null) {
+      out.nullValue();
+    } else {
+      String date = df.format(value);
+      out.value(date);
+    }
+  }
+
+  @Override public Date read(JsonReader in) throws IOException {
+    if (in.peek() == JsonToken.NULL) {
+      in.nextNull();
+      return null;
+    }
+    String stringDate = in.nextString();
+    try {
+      return df.parse(stringDate);
+    } catch (ParseException e) {
+      throw new DateTypeParseException(stringDate);
```

Comment:
I'd much rather not see a runtime exception in the sample, as in the real world this will cause the app to crash, and that's not what we want to encourage.  Instead, perhaps return a default value like `new Date()`